treeherder/model/models.py

```diff
@@ -136,6 +138,46 @@ class Meta:
     def __str__(self):
         return "{0}".format(self.id)
 
+    @classmethod
+    def search(cls, search_term):
+        max_size = 50
+        # 90 days ago
+        time_limit = datetime.datetime.now() - datetime.timedelta(days=90)
+        # Wrap search term so it is used as a phrase in the full-text search.
+        search_term_fulltext = search_term.join('""')
+        # Substitute escape and wildcard characters, so the search term is used
+        # literally in the LIKE statement.
+        search_term_like = search_term.replace('=', '==').replace(
+            '%', '=%').replace('_', '=_')
+        recent = cls.objects.raw(
+            '''
+            SELECT id, summary, crash_signature, keywords, os, resolution,
+            MATCH (`summary`) AGAINST (%s IN BOOLEAN MODE) AS relevance
+            FROM bugscache
+            WHERE 1
+              AND resolution = ''
+              AND `summary` LIKE CONCAT ('%%%%', %s, '%%%%') ESCAPE '='
+              AND modified >= %s
+            ORDER BY relevance DESC
+            LIMIT 0,%s
+            ''', [search_term_fulltext, search_term_like, time_limit,
+                  max_size])
+
+        all_others = cls.objects.raw(
+            '''
+            SELECT id, summary, crash_signature, keywords, os, resolution,
+            MATCH (`summary`) AGAINST (%s IN BOOLEAN MODE) AS relevance
+            FROM bugscache
+            WHERE 1
+            AND `summary` LIKE CONCAT ('%%%%', %s, '%%%%') ESCAPE '='
+            AND (modified < %s OR resolution <> '')
+            ORDER BY relevance DESC
+            LIMIT 0,%s''', [search_term_fulltext, search_term_like, time_limit,
+                            max_size])
+
+        return {"open_recent": [model_to_dict(item) for item in recent],
+                "all_others": [model_to_dict(item) for item in all_others]}
```

Comment:
Looks good.

When I came across some of the alternatives for converting to a dict, some of them added additional properties that wouldn't have been returned by say `.values_list()`, or in the case of `model_to_dict()` specifically, it doesn't return fields that cannot be used in a Django form context. Could you double check the returned dict still includes the expected keys? (perhaps in `test_bugscache.py`)